frappe/auth.py

```diff
@@ -237,15 +237,21 @@ def run_trigger(self, event='on_login'):
 
 	def validate_ip_address(self):
 		"""check if IP Address is valid"""
-		ip_list = frappe.db.get_value('User', self.user, 'restrict_ip', ignore=True)
+		ip_list = get_ip_list(self.user)
 		if not ip_list:
 			return
 
-		ip_list = ip_list.replace(",", "\n").split('\n')
-		ip_list = [i.strip() for i in ip_list]
-
+		bypass_restrict_ip_check = 0
+		# check if two factor auth is enabled
+		enabled = int(frappe.db.get_value('System Settings', None, 'enable_two_factor_auth') or 0)
```

Comment:
use `frappe.get_system_settings('enable_two_factor_auth)` it will cache and return

Reply:
@rmehta updated please check thanks